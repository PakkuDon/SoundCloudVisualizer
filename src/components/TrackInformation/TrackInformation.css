.root {
  display: flex;
  align-items: center;
  justify-content: center;
}

.thumbnail,
.details {
  display: inline-block;
}

.thumbnail {
  width: 100px;
  height: 100px;
  margin-right: 10px;
}

.details {
  text-align: left;
}
