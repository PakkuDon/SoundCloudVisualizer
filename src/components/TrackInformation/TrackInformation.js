import React from "react"
import PropTypes from "prop-types"
import { SoundCloudTrack } from "../../propTypes"

import styles from "./TrackInformation.css"

const TrackInformation = ({ track }) => (
  <div>
    <img className={styles.thumbnail} src={track.artwork_url} />
    <div className={styles.details}>
      <div>{track.user && track.user.username}</div>
      <div>{track.title}</div>
      <div>{track.genre}</div>
    </div>
  </div>
)

TrackInformation.propTypes = {
  track: SoundCloudTrack.isRequired,
}

export default TrackInformation
