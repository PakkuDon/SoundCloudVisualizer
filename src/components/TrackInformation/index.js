import TrackInformation from "./TrackInformation"
export default TrackInformation
