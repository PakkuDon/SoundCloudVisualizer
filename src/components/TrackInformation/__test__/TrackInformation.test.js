import React from 'react'
import { render } from "@testing-library/react"

import TrackInformation from '../TrackInformation'

const createTrack = () => ({
  id: 1,
  title: "Hello World",
  genre: "",
  permalink_url: "",
  stream_url: "",
  artwork_url: "",
  user: {
    username: "Bob",
  },
})

describe("TrackInformation", () => {
  it("renders track information", () => {
    const children = <div>Hello world</div>
    const { container } = render(<TrackInformation track={createTrack()} />)

    expect(container).toMatchSnapshot()
  })
})
