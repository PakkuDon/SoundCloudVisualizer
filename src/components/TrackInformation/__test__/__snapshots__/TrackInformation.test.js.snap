// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackInformation renders track information 1`] = `
<div>
  <div
    class="root"
  >
    <img
      alt="Hello World artwork"
      class="thumbnail"
      src=""
    />
    <div
      class="details"
    >
      <div>
        Bob
      </div>
      <div>
        Hello World
      </div>
      <div />
    </div>
  </div>
</div>
`;
